### PR TITLE
This PR fixes the issue #153 both on server and client side

### PR DESCRIPTION
**Pull request by levaja**

### CODE REVIEW

File: WatchDogServer/lib/watchdog_server.rb [hunk 1]
```diff
@@ -83,6 +83,11 @@ def mongo
   # Create a new user and return unique SHA1
   post '/user' do
     user = create_json_object(request)
+	
+	if user['programmingExperience'].nil? or user['programmingExperience'].empty?
+	  halt 404, "Wrong user registration"
```
Comment (Inventitech) [inline WatchDogServer/lib/watchdog_server.rb]: Can you specify a more precise error message (programming experience missing)?

Comment (levaja) [inline WatchDogServer/lib/watchdog_server.rb]: Fixed.

**Comment by levaja** [comment]
Test added :)

**Comment by Inventitech** [comment]
I get these errors when I run the tests:

➜  WatchDogServer git:(master) ✗ ./run_tests.sh 
17/17: _-*_*-_-_-_**-*_*_,------,   
11/17: _-*_*-_-_-_**-*_*_|  /\_/\ 
 0/17: _-*_*-_-_-_**-*_*~|_( x .x)  
 6/17: _-*_*-_-_-_**-*_* ""  "" 

Failures:

  1) The WatchDog Server should create a user when the details are correct
     Failure/Error: last_response.status.should eql(201)
       
       expected: 201
            got: 404
       
       (compared using eql?)
     # ./spec/watchdog_server_spec.rb:71:in `block (2 levels) in <top (required)>'

  2) The WatchDog Server should create a project when the details are correct
     Failure/Error: last_response.status.should eql(201)
       
       expected: 201
            got: 404
       
       (compared using eql?)
     # ./spec/watchdog_server_spec.rb:84:in `block (2 levels) in <top (required)>'

  3) The WatchDog Server should return 200 for existing user
     Failure/Error: get '/user/' + existing_user
     TypeError:
       can't convert nil into String
     # ./spec/watchdog_server_spec.rb:123:in `+'
     # ./spec/watchdog_server_spec.rb:123:in `block (2 levels) in <top (required)>'

  4) The WatchDog Server should return 200 for existing project
     Failure/Error: get '/project/' + existing_project
     TypeError:
       can't convert nil into String
     # ./spec/watchdog_server_spec.rb:128:in `+'
     # ./spec/watchdog_server_spec.rb:128:in `block (2 levels) in <top (required)>'

  5) The WatchDog Server should return 404 when posting intervals for non-existing project
     Failure/Error: post '/user/' + existing_user + '/intervals', intervals.to_json
     TypeError:
       can't convert nil into String
     # ./spec/watchdog_server_spec.rb:142:in `+'
     # ./spec/watchdog_server_spec.rb:142:in `block (2 levels) in <top (required)>'

  6) The WatchDog Server should return the number of stored intervals on successful insert
     Failure/Error: post "/user/#{user_id}/#{project_id}/intervals", intervals.to_json
     URI::InvalidURIError:
       bad URI(is not URI?): /user/Missing programming experience in user registration/The user who registers the project does not exist on the server. Create a new user first./intervals
     # ./spec/watchdog_server_spec.rb:160:in `block (2 levels) in <top (required)>'

You've Nyaned for 0.17 seconds

17 examples, 6 failures

Failed examples:

rspec ./spec/watchdog_server_spec.rb:68 # The WatchDog Server should create a user when the details are correct
rspec ./spec/watchdog_server_spec.rb:81 # The WatchDog Server should create a project when the details are correct
rspec ./spec/watchdog_server_spec.rb:122 # The WatchDog Server should return 200 for existing user
rspec ./spec/watchdog_server_spec.rb:127 # The WatchDog Server should return 200 for existing project
rspec ./spec/watchdog_server_spec.rb:139 # The WatchDog Server should return 404 when posting intervals for non-existing project
rspec ./spec/watchdog_server_spec.rb:151 # The WatchDog Server should return the number of stored intervals on successful insert

**Comment by levaja** [comment]
I get them as well, but they are from old tests (didn't want to touch that).

**Comment by levaja** [comment]
Oh, I am sorry, looking into this - maybe my edits in server code made some tests fail.

**Comment by levaja** [comment]
Okay, I've checked:

1) and 6) My fault, definition of valid user should now also have programmingExperience field (although it didn't exist previously).

2), 3), 4) and 5) are caused by setting existing_user and existing_project to nil, which I didn't change.

**Comment by Inventitech** [comment]
*All* failing tests are a regression introduced by this PR. (Revert to the preceding commit bdcc72f6f4acc2029e96d66d5395d027cc389fdf to verify this.)

**Comment by levaja** [comment]
Okay, I will look into this first thing in the morning, these tests are a bit new to me, so I don't want to rush like I did with my previous comments :)

**Comment by levaja** [comment]
Actually, I've checked now - if I'm correct, previous commit ("Fixed 2 broken tests") actually fixes all of 6 other tests, since 2, 3, 4 and 5 depend on success of 1.

Now all the tests are successful (although I had to comment the Send-registration-email command, since I don't actually send any emails.

**Comment by Inventitech** [comment]
Thanks for providing this PR!

**Pull request merged**